File: PHP/product.php
class Product{
  public string $name;
  public string $code;
  public int $price;

  public function __construct(string $name, string $code, int $price)
    {
        $this->name = $name; 
        $this->code = $code; 
        $this->price = $price; 
    }

  public function getName(): string 
    {
        return $this->name;
    }

  public function getCode(): string 
    {
        return $this->code;
    }

  public function getPrice(): string 
    {
        return $this->price;
    }
}

File: go/product.go
package product

type Product struct {
	code  string
	name  string
	price float64
}

func new_product(code string, name string, price float64) Product {
	return Product{code: code, name: name, price: price}
}

func (p Product) get_code() string {
	return p.code
}

func (p Product) get_name() string {
	return p.name
}

func (p Product) get_price() float64 {
	return p.price
}